src/Parser/Mage/Frost/Modules/Features/ThermalVoid.js

```diff
@@ -14,29 +14,38 @@ class ThermalVoid extends Module {
 	}
 
   casts = 0;
+  buffApplied = 0;
+  extraUptime = 0;
 
   on_initialized() {
 	   this.active = this.combatants.selected.hasTalent(SPELLS.THERMAL_VOID_TALENT.id);
   }
 
   on_toPlayer_applybuff(event) {
-    const spellId = event.ability.guid;
-    if(spellId === SPELLS.ICY_VEINS.id && event.prepull) { // catch prepull cast by buff being present on pull
+    if(event.ability.guid === SPELLS.ICY_VEINS.id && event.prepull) { // catch prepull cast by buff being present on pull
       this.casts += 1;
+      this.buffApplied = event.timestamp;
     }
   }
 
   on_byPlayer_cast(event) {
-    const spellId = event.ability.guid;
-    if (spellId !== SPELLS.ICY_VEINS.id) {
+    if (event.ability.guid !== SPELLS.ICY_VEINS.id) {
       return;
     }
     this.casts += 1;
+    this.buffApplied = event.timestamp;
   }
 
-  suggestions(when) {
-    const averageDuration = (this.combatants.selected.getBuffUptime(SPELLS.ICY_VEINS.id) / 1000) / this.casts;
+  on_finished() {
+    if (this.combatants.selected.hasBuff(SPELLS.ICY_VEINS.id)) {
+      this.casts -= 1;
+      this.extraUptime = this.owner.currentTimestamp - this.buffApplied;
```

Comment:
Is there a reason to remove the cast and not just show this as a shortened / clipped IV cast?

From an outsider perspective it seems like the cast in this case should count, however, tracking an additional metric of a clipped cast may make more sense.

Reply:
Yeah, it's because average Icy Veins uptime is an important frost mage performance metric (it works a bit similar to spriests void form, but is available much less frequently). If the fight ends with IV still up, it may come out as a too short uptime, even though this is not the player's fault, and it can throw off their overall metric. For this reason, we want to avoid including IV that is still up when fight ends.

Reply:
Gotcha. Sounds good. Just wanted to ask the question. 😃